### PR TITLE
Convert a few simple components to use hooks

### PR DESCRIPTION
**Pull request by BPScott**
*This PR is best viewed if you disable whitespace changes, as the hookification saves a level of indentation, which makes it look like i changed EVERYTHING*

### WHY are these changes introduced?

Starting down the path of removing withAppProvider usage into the new world of hooks.
Part of #1995

### WHAT is this pull request doing?

Remove withAppProvider, and use hooks for the following components:

ActionMenu/RollupAction
Autocomplete
Card
EmptySearchResult
Form
SkeletonPage
TopBar

### How to 🎩

- ActionMeni/RollupAction: Confirm ... popover on with Page with action group example at small screen sizes renders and toggles on click
- Autocomplete: Check interactions on any Autocomplete example
- Card: Confirm secondary actions popover still works on Card
- EmptySearchResult: Check `<EmptySearchResult title="Title" description="Description" withIllustration />` renders in playground
- Form: Check any form example
- SkeletonPage: Check and skeleton page example
- Topbar: check navigation visibility toggling works in Frame example at small screen sizes

### CODE REVIEW

File: src/utilities/use-toggle.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export function useToggle(initialState: boolean) {
+  const [state, setState] = useState(initialState);
+  const toggle = useCallback(() => setState((state) => !state), []);
+
+  // cast needed to say this returns a two item array with the items in
+  // their specific positions instead of `(typeof state | typeof toggle)[]`
+  return [state, toggle] as [typeof state, typeof toggle];
+}
+
+export function useForcableToggle(initialState: boolean) {
```
Comment (amrocha) [inline src/utilities/use-toggle.ts]: I think this is a bit of a stretch, I would just do this manually in the file itself and I don't think it comes up often enough to warrant a custom hook. Idk, feels a bit weird, if others think it's fine I'm fine with it though

Comment (BPScott) [reply src/utilities/use-toggle.ts]: I've saw cases where we want to force set a boolean toggle to false (e.g. when you click a modal's close button / leave focus in popovers.

I think this crops up often enough to make a general purpose thing.

File: src/utilities/use-toggle.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import {useState, useCallback} from 'react';
+
+export function useToggle(initialState: boolean) {
```
Comment (dleroux) [inline src/utilities/use-toggle.ts]: I can see why you did this, I just wonder it's worth it. It's simple but it requires knowledge about the hook and it's less declarative and just handling the toggle when needed. We aren't saving on much here imo.

Comment (BPScott) [reply src/utilities/use-toggle.ts]: The same can be said for pretty much all of our custom hooks :)

IMO The value from these small utilities comes from encapsulation, that small amount of complexity saved adds up over the course of several components and its repeated presence means we'll become familiar with it over time for when we have a boolean toggle, which is pretty often.

Comment (BPScott) [reply src/utilities/use-toggle.ts]: The same can be said for pretty much all of our custom hooks :)

IMO The value from these small utilities comes from encapsulation, that small amount of complexity saved adds up over the course of several components and its repeated presence means we'll become familiar with it over time for when we have a boolean toggle, which is pretty often.

Comment (amrocha) [reply src/utilities/use-toggle.ts]: I think for a really common usecase like a toggle variable it's fine. I agree that every hook adds to the cognitive load needed to understand the codebase, and we should avoid them

**Pull request merged**